pytorch_lightning/trainer/progress.py

```diff
@@ -151,7 +177,7 @@ class EpochProgress(Progress):
         current: Tracks the current epoch progress
         batch: Tracks batch progress.
     """
-
+    dataloader_idx: int = 0
```

Comment:
we have a dataloader_idx here but also in the special DataLoaderProgress?
Is this intentional?

Reply:
Yes, some dataclasses there are legacy and need to be dropped.